tests/unit/NetworkTest.js

```diff
@@ -209,3 +221,45 @@ test('consecutive API calls eventually succeed when authToken is expired', () =>
             expect(chatList).toEqual(TEST_CHAT_LIST);
         });
 });
+
+test('retry network request if auth and credentials are not read from Onyx yet', () => {
+    Onyx.addDelayToConnectCallback(3000);
+
+    // Reset isReady for this unit test, because Onyx.clear() in beforeEach resolves the
+    // Onyx.connect callbacks with null values
```

Comment:
I think it would be good to adopt the `GIVEN` `WHEN` `THEN` style of comments here. There's some information on that here:

https://github.com/Expensify/App/blob/main/tests/README.md#documenting-tests